third_party/docfx-doclet-143274/src/main/java/com/microsoft/lookup/PackageLookup.java

```diff
@@ -51,4 +66,112 @@ public String extractJavaType(PackageElement element) {
     }
     return null;
   }
+
+  /** Compare PackageElements by their parsed ApiVersion */
+  private final Comparator<PackageElement> byComparingApiVersion =
+      Comparator.comparing(pkg -> extractApiVersion(pkg).orElse(ApiVersion.NONE));
+
+  public Optional<ApiVersion> extractApiVersion(PackageElement pkg) {
+    String name = String.valueOf(pkg.getQualifiedName());
+    int lastPackageIndex = name.lastIndexOf('.');
+    String leafPackage = name.substring(lastPackageIndex + 1);
+    return ApiVersion.parse(leafPackage);
+  }
+
+  public enum PackageGroup {
+    VISIBLE,
+    OLDER_AND_PRERELEASE
+  }
+
+  /**
+   * Organize packages into PackageGroups, making some VISIBLE the rest hidden under the
+   * OLDER_AND_PRERELEASE category.
+   */
+  public ImmutableListMultimap<PackageGroup, PackageElement> organize(
+      List<PackageElement> packages) {
+
+    ListMultimap<PackageGroup, PackageElement> organized =
+        MultimapBuilder.enumKeys(PackageGroup.class).arrayListValues().build();
+
+    Multimap<String, PackageElement> packagesGroups = groupVersions(packages);
+    ImmutableList<String> alphabetizedPackageGroups =
+        packagesGroups.keySet().stream().sorted().collect(ImmutableList.toImmutableList());
+
+    for (String name : alphabetizedPackageGroups) {
+      Collection<PackageElement> versions = packagesGroups.get(name);
+
+      // The recommended package of each group is made visible.
+      PackageElement recommendedVersion = getRecommended(versions);
+      organized.put(PackageGroup.VISIBLE, recommendedVersion);
+
+      // All others are added to "Older and prerelease versions"
+      versions.stream()
+          .filter(version -> !version.equals(recommendedVersion))
+          .sorted(byComparingApiVersion)
+          .forEach(
+              version -> {
+                organized.put(PackageGroup.OLDER_AND_PRERELEASE, version);
+              });
+    }
+
+    return ImmutableListMultimap.copyOf(organized);
+  }
+
+  /**
+   * This 'grouping' logic combines all versioned packages together in a single `a.b.c.v#` group.
+   *
+   * <p>For example: a.b.v1 and a.b.v2 will be in the same group, but a.b and a.b.c will be in their
+   * own groups.
+   *
+   * <p>When packages are grouped, only one package within the group will be VISIBLE and the rest
+   * will be placed in the OLDER_AND_PRERELEASE category.
+   */
+  @VisibleForTesting
+  Multimap<String, PackageElement> groupVersions(List<PackageElement> packages) {

```

Comment:
IIUC, does this mean that there is one recommended package per stable version? So if an API has v1, v1alpha, v2, v2beta1, then v1 and v2 would be visible and v1alpha and v2beta1 would be hidden under the "Older and Prerelease packages"?

Reply:
One version will be visible, and the rest will be hidden under "Older and prerelease packages". 
Ex: If v1 and v2 exist, only v2 is visible and v1 is under "Older..."
Ex: If v1 and v2beta exists, only v1 is visible and v2beta is under "Older and prerelease..."